redirects/models.py

```diff
@@ -132,7 +131,8 @@ def get_staff_delete_url(self):
     @property
     def obj(self):
         """Work out which object we're pointing to"""
-        return first([getattr(self, f.name, None) for f in self.targets()])
+        field_names = [getattr(self, f.name, None) for f in self.targets()]
+        return next((f for f in field_names if f), None)  # pragma: no branch
```

Comment:
thought: This could perhaps complain more loudly, instead of returning `None`.

Do we ever expect it could not find a match?

(In other code, we access attributes of `.obj()`. For example, [in `type()` below](https://github.com/opensafely-core/job-server/pull/4503/files#diff-b4dca3e75ff654d499fcd477fafbf5671808e7180271f9f3bbd1f6e91172c75cR152-R153).)

Reply:
Still not entirely sure about this. No big harm in leaving it, but not sure.

Reply:
I think the default's removable; the only reference to this code is in `Redirect.type()` which accesses attributes, as I said, so the implicit assumption is that it's a non-falsy value.